cinema/permitions.py

```diff
@@ -0,0 +1,40 @@
+from rest_framework.permissions import BasePermission, SAFE_METHODS
+
+
+class IsAdminOrIfAuthenticatedReadOnly(BasePermission):
+    """
+    The request is authenticated as a admin_user, or is a read-only request.
+    """
+
+    def has_permission(self, request, view):
+        return bool(
+            (
+                request.user
+                and request.user.is_authenticated
+                and request.method in SAFE_METHODS
+            )
+            or (request.user and request.user.is_staff)
+        )
+
+
+class OrderPermission(BasePermission):
+    """
+    The authenticated user can create. And only owner can observe
+    """
+
+    def has_permission(self, request, view):
+
+        return bool(
+            (request.method == "POST" or request.method in SAFE_METHODS)
+            and request.user
+            and (request.user.is_authenticated or request.user.is_staff)

```

Comment:
The logic in `has_permission` for `OrderPermission` seems to allow any authenticated user to create or perform safe methods, but it doesn't restrict observation to only the owner as described in the class docstring. You might want to ensure that only the owner can observe, which should be reflected in the `has_object_permission` method.